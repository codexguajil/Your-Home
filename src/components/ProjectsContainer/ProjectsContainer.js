import React, { Component } from 'react';
// import mockProjects from '../../mockProject';
import Project from '../Project/Project';

export class ProjectsContainer extends Component {
  
  renderProjects = () => {
    return this.props.projects.map((project, i) => {
      return <Project key={i} {...project}/>
    });
  }

  render() {
    return (
      <div className='projects'>
        {this.renderProjects()}
      </div>
    )
  }
}

export default ProjectsContainer
