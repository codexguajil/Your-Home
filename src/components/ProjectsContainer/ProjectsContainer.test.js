import React from 'react';
import { shallow } from 'enzyme';
import ProjectsContainer from './ProjectsContainer';
import mockProjects from '../../mockProject'

describe('ProjectsContainer', () => {
  let wrapper;
  let projects;
  beforeEach(() => {
    projects = mockProjects.data.projects
    wrapper = shallow(<ProjectsContainer projects={projects}/>)
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});