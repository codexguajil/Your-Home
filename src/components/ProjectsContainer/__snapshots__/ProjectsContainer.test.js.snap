// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectsContainer should match snapshot 1`] = `
<div
  className="projects"
>
  <Project
    address="123 Fake St."
    description="Big, white house"
    id={1}
    key="0"
    name="House 1"
    rooms={
      Array [
        Object {
          "description": "Northeast living room",
          "name": "Living Room 1",
          "roomMaterials": Array [
            Object {
              "element_type": "Flooring",
              "id": 1,
              "material": Object {
                "brand": "Kenmoore",
                "id": 1,
                "manual_url": null,
                "model_number": "abc1",
                "name": "Material 1",
                "notes": null,
                "quantity": null,
                "unit_price": null,
                "vendor": "HD",
              },
            },
            Object {
              "element_type": "Flooring",
              "id": 2,
              "material": Object {
                "brand": "Kenmoore",
                "id": 2,
                "manual_url": null,
                "model_number": "abc2",
                "name": "Material 2",
                "notes": null,
                "quantity": null,
                "unit_price": null,
                "vendor": "HD",
              },
            },
            Object {
              "element_type": "Fixtures",
              "id": 3,
              "material": Object {
                "brand": "GE",
                "id": 3,
                "manual_url": null,
                "model_number": "abc1",
                "name": "Material A",
                "notes": null,
                "quantity": null,
                "unit_price": null,
                "vendor": "HD",
              },
            },
          ],
          "type": "Living Room",
        },
        Object {
          "description": "Big Kitchen",
          "name": "Room 2",
          "roomMaterials": Array [
            Object {
              "element_type": "Shower",
              "id": 3,
              "material": Object {
                "brand": "Kenmoore",
                "id": 1,
                "manual_url": null,
                "model_number": "abc1",
                "name": "Material 1",
                "notes": null,
                "quantity": null,
                "unit_price": null,
                "vendor": "HD",
              },
            },
          ],
          "type": "Kitchen",
        },
      ]
    }
  />
  <Project
    address="456 Random Lane"
    description="Small with picket fence"
    id={2}
    key="1"
    name="House 2"
    rooms={
      Array [
        Object {
          "description": "Small bedroom",
          "name": "Room 3",
          "roomMaterials": Array [],
          "type": "Bedroom",
        },
      ]
    }
  />
</div>
`;
